### DISCUSSION TITLE
Sending custom headers with apache-airflow[otel]

### DISCUSSION THREAD

**Discussion by ninad-stackgen** [apache/airflow#42118] [Q&A]
I am running the AWS Apache Airflow instance locally using [aws-mwaa-local-runner](https://github.com/aws/aws-mwaa-local-runner). 

I am configuring the airflow configuration file (airflow.cfg) for sending metrics to a specific otlp endpoint (I have an open telemetry collector instance running in a kubernetes cluster). I have configured the metrics section as follows 

```
[metrics]
otel_on = True
otel_host = <my_otlp_endpoint>
otel_port = 4318
otel_ssl_active = False
```

I need to pass the Basic auth via headers. But I dont see an option for passing custom headers in the [metrics specific configuration options](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#metrics).

Is there a way to send custom headers for the above mentioned purpose?

**Comment by Kludex** [comment]
I'm also interested in this.

**Comment by potiuk** [comment]
So maybe someone would like to contribute it if there is no other option? @ninad-opsverse @Kludex ? I have no idea if it's possible but deep dive in the code should give the answer and maybe it would be good to add such feature.

**Reply by Kludex** [reply]
Starting point: https://github.com/apache/airflow/pull/44258
